src/module.ts

```diff
@@ -0,0 +1,147 @@
+import { promises as fsp } from 'fs'
+import { join, resolve } from 'pathe'
+import template from 'lodash.template'
+import { addPlugin, addTemplate, defineNuxtModule, isNuxt2, addComponent, addAutoImport } from '@nuxt/kit'
+import { fileURLToPath } from 'url'
+
+import { name, version } from '../package.json'
+
+const DEFAULTS: ModuleOptions = {
+  preference: 'system',
+  fallback: 'light',
+  hid: 'nuxt-color-mode-script',
+  globalName: '__NUXT_COLOR_MODE__',
+  componentName: 'ColorScheme',
+  classPrefix: '',
+  classSuffix: '-mode',
+  storageKey: 'nuxt-color-mode'
+}
+
+export default defineNuxtModule({
+  meta: {
+    name,
+    version,
+    configKey: 'colorMode',
+  },
+  defaults: DEFAULTS,
+  async setup (options, nuxt) {
+    // Read script from disk and add to options
+    const scriptPath = fileURLToPath(new URL('./script.min.js', import.meta.url))
```

Comment:
Could probably simplify now with @pi0's new helper `createResolver` - https://github.com/nuxt/framework/pull/3111 here and below where `import.meta.url` is referenced.